### PR TITLE
TMA Implementation

### PR DESCRIPTION
**Pull request by fdelcampo**
This PR contains TMA Implementation.

### CODE REVIEW

File: love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+import {
+    mtMountMirrorCoversStateMap,
+    stateToStyleMTMountMirrorCoversState,
+  } from '../../../../Config';
+import styles from './MirrorCovers.module.css';
+
+
+export default class MirrorCovers extends Component {
+  static propTypes = {
+    /** Mirror Covers view width */
```
Comment (sebastian-aranda) [inline love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx:14]: Good job 👍️

File: love/src/components/MainTel/TMA/Elevation/Elevation.jsx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+  render() {
+    return (
+        <div className={styles.container}>    
+            { this.getSvg(this.props)}
+        </div>
+    );
+  }
+
+  getSvg = (props) => {
+    const { width, height } = this.props;
```
Comment (sebastian-aranda) [inline love/src/components/MainTel/TMA/Elevation/Elevation.jsx:52]: I think this variables are not called anywere so they should be removed

**Review by sebastian-aranda — CHANGES_REQUESTED** [review]
Great job here Fernando! I some comments regarding some details, there rest looks very good

File: love/src/Config.js [hunk 3]
```diff
@@ -343,6 +343,76 @@ export const motorDriveStateMap = {
...
+};
+
+export const mtMountCommanderStateMap = {
+  0: 'NONE',
+  1: 'CSC',
+  2: 'EUI',
+  3: 'HHD',
+};
+
+// TODO: Pending Style
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/Config.js:361]: Yes, need validation about the colors used for the states.
@MiaRoseElbo

File: love/src/redux/selectors/selectors.js [hunk 4]
```diff
@@ -660,3 +660,129 @@ export const getLastAlarm = (state) => {
...
+ * @param {object} state
+ */
+ export const getMirrorCoversMotionState = (state) => {
+  const subscriptions = [
+    'event-MTMount-0-mirrorCoversMotionState',
+  ];
+  const summaryData = getStreamsData(state, subscriptions);
+  return {
+    mirrorCovers: summaryData['event-MTMount-0-mirrorCoversMotionState']
+      ? summaryData['event-MTMount-0-elevationSystemState'][0].state.value
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/redux/selectors/selectors.js]: thanks

File: love/src/redux/selectors/selectors.js [hunk 5]
```diff
@@ -660,3 +660,129 @@ export const getLastAlarm = (state) => {
...
+    trackID: summaryData['event-MTMount-0-target']
+      ? summaryData['event-MTMount-0-target'][0].trackId.value
+      : '',
+    commander: summaryData['event-MTMount-0-commander']
+      ? summaryData['event-MTMount-0-commander'][0].commander.value
+      : 0,
+    connected: summaryData['event-MTMount-0-connected']
+      ? summaryData['event-MTMount-0-connected'][0].command.value
+      : false,
+    balancing: summaryData['event-MTMount-0-commander']
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/redux/selectors/selectors.js:686]: thanks

File: love/src/components/MainTel/TMA/TMA.module.css [hunk 6]
```diff
@@ -0,0 +1,36 @@
+.summaryContainer {
+    display: flex;
+    flex-grow: 1;
+    flex-direction: column;
+    padding: 1em;
+    border-radius: 5px;
+    /* background: #132631; svg color, slightly better contrast with background */
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/components/MainTel/TMA/TMA.module.css:7]: yes, for the table, its not necessary. padding and border yes

File: love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+    super(props);
+    this.prevAzimuthActual = 0;
+    this.prevAzimuthDemand = 0;
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.azimuthActualPosition !== this.props.azimuthActualPosition)
+      this.prevAzimuthActual = this.closestEquivalentAngle(this.prevAzimuthActual, prevProps.azimuthActualPosition);
+    if (prevProps.azimuthDemandPosition !== this.props.azimuthDemandPosition)
+      this.prevAzimuthDemand = this.closestEquivalentAngle(this.prevAzimuthDemand, prevProps.azimuthDemandPosition);
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx:42]: OK, Its changed

File: love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+    const offset = 10;
+    const viewBoxSize = 385 - 2 * offset;
+    const x0 = viewBoxSize / 2 + offset;
+    const y0 = viewBoxSize / 2 + offset;
+
+    const angleClosed = this.getAngleClosedCoverMirror();
+    const angleClosedBorder = this.getAngleClosedCoverMirrorBorder();
+    
+    const equivalentAzimuthActual = this.closestEquivalentAngle(this.prevAzimuthActual, this.props.azimuthActualPosition);
+    const equivalentAzimuthDemand = this.closestEquivalentAngle(this.prevAzimuthDemand, this.props.azimuthDemandPosition);
```
[the remark this answers is not in the document]
Comment (fdelcampo) [reply love/src/components/MainTel/TMA/MirrorCovers/MirrorCovers.jsx:95]: Ok, its changed

File: love/src/components/MainTel/TMA/Summary/Summary.jsx [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+        const trackID = this.props.trackID ? this.props.trackID : '' ;
+        const commanderValue = this.props.commander ? mtMountCommanderStateMap[this.props.commander] : mtMountCommanderStateMap[0];
+        const connected = this.props.connected ? true : false;
+        const balancingValue = this.props.balancing ? mtMountPowerStateMap[this.props.balancing] : mtMountPowerStateMap[0];
+
+        const azimuthSystemValue = this.props.azimuthSystem ? mtMountPowerStateMap[this.props.azimuthSystem] : mtMountPowerStateMap[0];
+        const azimuthMotionValue = this.props.azimuthMotion ? mtMountAxisMotionStateMap[this.props.azimuthMotion] : mtMountAxisMotionStateMap[0];
+        const azimuthActualPosition = this.props.azimuthActualPosition ? this.props.azimuthActualPosition : 0;
+        const azimuthDemandPosition = this.props.azimuthDemandPosition ? this.props.azimuthDemandPosition : 0;
+        const azimuthLimits = this.azimuthLimits ? this.azimuthLimits : 0;
```
[the remark this answers is not in the document]
Comment (areyesd14) [reply love/src/components/MainTel/TMA/Summary/Summary.jsx:81]: is this resolved?

File: love/src/components/MainTel/TMA/Summary/Summary.jsx [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+
+        // AzimuthLimit
+        const minAzimuthPosition = 0;
+        const maxAzimuthPosition = 360;
+        const timeToAzimuthLimit = 0;
+        // const closestAzimuthLimit = 90;
```
[the remark this answers is not in the document]
Comment (areyesd14) [reply love/src/components/MainTel/TMA/Summary/Summary.jsx:87]: I think this is unresolved.

File: love/src/components/MainTel/TMA/Summary/Summary.jsx [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+
+        const elevationSystemValue = this.props.elevationSystem ? mtMountPowerStateMap[this.props.elevationSystem] : mtMountPowerStateMap[0];
+        const elevationMotionValue = this.props.elevationMotion ? mtMountAxisMotionStateMap[this.props.elevationMotion] : mtMountAxisMotionStateMap[0];
+        const elevationActualPosition = this.props.elevationActualPosition ? this.props.elevationActualPosition : 0;
+        const elevationDemandPosition = this.props.elevationDemandPosition ? this.props.elevationDemandPosition : 0;
+        const elevationLimits = this.elevationLimits ? this.elevationLimits : 0;
```
[the remark this answers is not in the document]
Comment (areyesd14) [reply love/src/components/MainTel/TMA/Summary/Summary.jsx:93]: I think this is unresolved.

**Review by areyesd14 — APPROVED** [review]
If all Sebastian comments were resolved, I approve this PR. Excellent job @fdelcampo !!

**Review by sebastian-aranda — APPROVED** [review]
Everything looks good and ready for merge! Great job @fdelcampo

**Pull request merged**